### ISSUE TITLE
Improve/Create build process for lib

### ISSUE THREAD

**Issue by dalestone** [dalestone/angular2-gantt#26]

**Comment by 11joselu** [comment]
angular-cli current situation for build [components as library](https://github.com/angular/angular-cli/issues/1624#issuecomment-279443396)


Reference to feature (boilerplate for library): [Feature Create boilerplate for library · Issue #1692 · angular/angular-cli](https://github.com/angular/angular-cli/issues/1692)

## To do

- [x] Migration from Webpack build system to Gulp
- [x] Create a new scripts in package.json
- [x] Create a new tsconfig.json
- [x] Create a configuration for typescript lint (tslint.json) 
- [ ] Create and configure .gitignore, .npminore etc.
- [x] Create a main file
- [ ] Default export for compatibility with angular cli [see](https://github.com/angular/angular-cli/issues/96)
- [x] Add support for types
- [ ] ~~Webpack configuration to show demo page.~~
- [ ] Css/Sass configuration
- [x] Template import or bundle
- [ ] ~~Exclude external libs from main file~~
- [ ] Build for system compatibility
- [ ] Build for cjs compatibility 
- [ ] e2e scripts
- [ ] Test scripts

I will keep updating this list ~~Scratch this.~~

**Comment by 11joselu** [comment]
Brief summary.

_First of all I am sorry for possibles English errors (not my native language)_

I tried to adapt Webpack configuration for bundle multiples files, inlines resources and compile sass files into separated files,  with no great results. After looking some plugins and theirs architecture, I changed from  Webpack to Gulp.js using Typescript. 

[Working repo](https://github.com/11joselu/ng-library-boilerplate)

This build process, compile TS files into CommonJs module format, inject inline resources (Sass (compiled), HTML), and create a build folder with all js files and css (currently I'm not copying all src files).

**What next's?**

- Working on testing
